### PR TITLE
Boolean checkbox widget

### PR DESCRIPTION
**Pull request by rsanzante**
Shaped from current StringTextfield widget.

https://github.com/jsdrupal/drupal-admin-ui/issues/156

Not really sure this what is wanted.

### CODE REVIEW

**Comment by rsanzante** [comment]
Thanks for your review! I think I've included it.

**Comment by rsanzante** [comment]
Oops, conflict resolved but now eslint complains about prettier errors. FIxing them (although I think I got them during master merge to resolve conflict).

**Comment by rsanzante** [comment]
Solved. I wonder if eslint properly configured because some changes seem ugly to me. It forced me to put multiline declaration into one line harder to read. See my last commit.

**Comment by dawehner** [comment]
It doesn't work properly right now. We need something like this at least I believe

```
diff --git a/src/components/02_atoms/Widgets/BooleanCheckbox.js b/src/components/02_atoms/Widgets/BooleanCheckbox.js
index a5bd85e..5fec7c3 100644
--- a/src/components/02_atoms/Widgets/BooleanCheckbox.js
+++ b/src/components/02_atoms/Widgets/BooleanCheckbox.js
@@ -6,8 +6,8 @@ import WidgetPropTypes from '../../05_pages/NodeForm/WidgetPropTypes';
 const BooleanCheckbox = props => (
   <CheckBox
     id={props.fieldName}
-    checked={props.checked}
-    onChange={event => props.onChange(event.target.value)}
+    checked={props.value}
+    onChange={event => props.onChange(event.target.checked)}
     margin="normal"
     label={props.label}
   />
@@ -15,12 +15,12 @@ const BooleanCheckbox = props => (
 
 BooleanCheckbox.propTypes = {
   ...WidgetPropTypes,
-  checked: PropTypes.bool,
+  value: PropTypes.bool,
   label: PropTypes.string,
 };
 
 BooleanCheckbox.defaultProps = {
-  checked: false,
+  value: false,
   label: 'On',
 };
 

```

File: src/components/02_atoms/Widgets/BooleanCheckbox.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+import CheckBox from '@material-ui/core/Checkbox';
+import WidgetPropTypes from '../../05_pages/NodeForm/WidgetPropTypes';
+
+const BooleanCheckbox = props => (
+  <CheckBox
+    id={props.fieldName}
+    checked={props.checked}
+    onChange={event => props.onChange(event.target.value)}
+    margin="normal"
+    label={props.label}
```
Comment (martinfrances107) [inline src/components/02_atoms/Widgets/BooleanCheckbox.js]: label is not a valid property here 

can I suggest wrapping checkbox in FormControlLabel
I am also thinking about how to make this extendable and want to pass any extra properties like color="primary" down into the checkbox element.

I hope this helps

  render() {
    /* Pass all props not used at this level down into Checkbox */
    const { onChange, label, ...checkboxProps } = this.props;
    return (
      <FormControlLabel
        control={
          <Checkbox
            {...checkboxProps}
            checked={this.state.checked}
            onChange={onChange}
            value=""
          />
        }
        label={label}
      />
    );
  }

Comment (dawehner) [reply src/components/02_atoms/Widgets/BooleanCheckbox.js]: Great remark @martinfrances107 !

**Comment by rsanzante** [comment]
Fixes:
  - Wrap CheckBox with a FormControlLabel.
  - Pass addtional props to CheckBox.
  - Fix value/checked prop.
  - Added id to both elements using fieldName and a suffix (-label and -cb).

**Comment by francescopersico** [comment]
Yes, what i mean is that the only files included in this PR should be:

BooleanCheckbox.js
UiMetadata.js
Widgets.js

Other files are not related to the PR, and if they need formatting/linting fixes should be done outside this PR.

**Comment by dawehner** [comment]
@francescopersico Oops I think I misread your message on my phone :)

File: src/components/02_atoms/Widgets/BooleanCheckbox.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+const BooleanCheckbox = props => {
+  const { onChange, label, id, value, ...checkboxProps } = props;
+
+  return (
+    <FormControlLabel
+      id={`${props.fieldName}-label`}
+      control={
+        <CheckBox
+          id={`${props.fieldName}-cb`}
+          onChange={event => onChange(event.target.value)}
```
Comment (dawehner) [inline src/components/02_atoms/Widgets/BooleanCheckbox.js]: Note: for setting the right checked value, you need to access ```event.target.checked```:

```
event.target.value
""
event.target.checked
true
```

Comment (rsanzante) [reply src/components/02_atoms/Widgets/BooleanCheckbox.js]: Interesting, ```value``` returns "true" or "", while ```checked``` returns ```true``` or ```false```.

Changed!

**Comment by dawehner** [comment]
@rsanzante Great work!

**Comment by dawehner** [comment]
Do you want to come to our initiative meeting later in #javascript ?

**Pull request merged**